### PR TITLE
ruler first draft

### LINKED ISSUE #8: Create a ruler tool
Create a ruler that people can use to measure distances.

### PR DESCRIPTION
**Pull request by regexbuster**
This has some funky new characters which provides a little ruler thing. Not sure how we're measuring distance so I just had it display euclidean distance between the two endpoints based on the canvas position. Cheers!

### CODE REVIEW

**Comment by regexbuster** [comment]
image

**Comment by regexbuster** [comment]
I've added some fancy characters that makes the ruler not leave the canvas if you try to pull it off.

**Comment by AjaiKN** [comment]
The file now has mixed tabs and spaces, so I converted it all to tabs since that's what it was originally in this file. (I'm guilty of making the tabs/spaces inconsistent between files a while ago, but we can fix that later.)

**Comment by regexbuster** [comment]
I end up copying things from other files and moving stuff around and the spaces and tabs get all messed up often so sorry about that. Also, should I merge it?

File: script.js [hunk 1]
```diff
@@ -97,10 +97,99 @@ class CircleMover extends Mover {
...
+		text(String(dist),(this.shape1.x + this.shape2.x)/2-15,(this.shape1.y + this.shape2.y)/2-15)
+	}
+}
+
+
 let allObjects;
 let isMouseBeingPressed = false;
 let paused = false;
-let canvas;
+let canvas, r;
```
Comment (AjaiKN) [inline script.js]: Could we use a more descriptive name? Probably `ruler`?

File: script.js [hunk 2]
```diff
@@ -97,10 +97,99 @@ class CircleMover extends Mover {
 	}
 }
 
+class Draggable {
+	constructor(x,y,w,h){
+		this.dragging = false;
+		this.rollover = false;
```
Comment (AjaiKN) [inline script.js]: What does `rollover` mean? A more descriptive name might be helpful if you can think of one.

File: script.js [hunk 3]
```diff
@@ -97,10 +97,99 @@ class CircleMover extends Mover {
...
+		this.x = x;
+		this.y = y;
+		this.w = w;
+		this.h = h;
+		this.offsetX = 0;
+		this.offsetY = 0;
+	}
+
+	over(){
```
Comment (AjaiKN) [inline script.js]: More descriptive name for `over`?

**Review by AjaiKN — CHANGES_REQUESTED** [review]

**Comment by AjaiKN** [comment]
Once we change those variable names, I'm good with merging. Thanks for the PR!

File: script.js (hunk 1 above)
Comment (regexbuster) [reply script.js]: Oops, been needing to conserve characters for a programming game recently and happened to condense the variable name. I'll fix these few changes real quick.

Comment (AjaiKN) [reply script.js]: No worries!

File: script.js (hunk 2 above)
Comment (regexbuster) [reply script.js]: Changed to mouseOver as it's for changing come colors when you are moused over an endpoint.

File: script.js (hunk 3 above)
Comment (regexbuster) [reply script.js]: Changed to mousedOver(). Slightly more descriptive I guess.

**Comment by regexbuster** [comment]
Was there anything else we wanted for the ruler?
Currently this has:
- Ruler
- Toggle Button
- Knobs with circular bounding boxes
- Length displayed in pixels
- Ruler resets when button is pressed
(I assume we can come back to this when we figure out how we want to do actual distances)

**Comment by AjaiKN** [comment]
I agree that we can do distances later. I'll merge.

**Review by AjaiKN — APPROVED** [review]

**Pull request merged**